File: cpp/232_ImplementQueueusingStacks.cpp
class Queue {
public:
    // Push element x to the back of queue.
    void push(int x) {
        inStack.push(x);
    }

    // Removes the element from in front of queue.
    void pop(void) {
        peek();
        outStack.pop();
    }

    // Get the front element.
    int peek(void) {
        if(outStack.empty()){
            while(inStack.size()){
                outStack.push(inStack.top());
                inStack.pop();
            }
        }
        return outStack.top();
    }

    // Return whether the queue is empty.
    bool empty(void) {
        return inStack.empty() && outStack.empty();
    }
private:
    stack<int> inStack, outStack;
};

File: js/232_ImplementQueueusingStacks.js
/**
 * @constructor
 */
 
var Queue = function() {
    this.in = [];
    this.out = [];
};

/**
 * @param {number} x
 * @returns {void}
 */
Queue.prototype.push = function(x) {
    this.in.push(x);
};

/**
 * @returns {void}
 */
Queue.prototype.pop = function() {
    this.peek();
    this.out.pop();
};

/**
 * @returns {number}
 */
Queue.prototype.peek = function() {
    if(this.out.length < 1){
        while(this.in.length > 0){
            this.out.push(this.in.pop());
        }
    }
    return this.out[this.out.length - 1];
};

/**
 * @returns {boolean}
 */
Queue.prototype.empty = function() {
    return this.in.length < 1 && this.out.length < 1;
};